### PR TITLE
x11-*: update various packages

### PR DESCRIPTION
**Pull request by ElectrodeYT**
These updates seem to fix X11 for me.

### CODE REVIEW

File: bootstrap.d/x11-libs.yml [hunk 1]
```diff
@@ -503,8 +497,8 @@ packages:
     source:
       subdir: ports
       git: 'https://gitlab.freedesktop.org/xorg/lib/libx11.git'
-      tag: 'libX11-1.8.1'
-      version: '1.8.1'
+      tag: 'libX11-1.8.6'
```
Comment (Dennisbonke) [inline bootstrap.d/x11-libs.yml:500]: This port does not compile without a newer autoconf, I propose downgrading to `1.8.4`.

File: bootstrap.d/x11-libs.yml [hunk 2]
```diff
@@ -1068,8 +1051,8 @@ packages:
     source:
       subdir: ports
       git: 'https://gitlab.freedesktop.org/xorg/lib/libxi.git'
-      tag: 'libXi-1.8'
-      version: '1.8'
+      tag: 'libXi-1.8.1'
```
Comment (Dennisbonke) [inline bootstrap.d/x11-libs.yml:1054]: Port did not compile, I propose not upgrading (failing in regenerate).

File: bootstrap.d/x11-libs.yml [hunk 3]
```diff
@@ -1301,8 +1279,8 @@ packages:
     source:
       subdir: ports
       git: 'https://gitlab.freedesktop.org/xorg/lib/libxpm.git'
-      tag: 'libXpm-3.5.13'
-      version: '3.5.13'
+      tag: 'libXpm-3.5.16'
```
Comment (Dennisbonke) [inline bootstrap.d/x11-libs.yml:1282]: Port did not compile, I propose moving to 3.5.14.

File: bootstrap.d/x11-libs.yml [hunk 4]
```diff
@@ -67,37 +66,35 @@ packages:
...
-          # freetype-config does not support cross-compilation.
-          FREETYPE_CONFIG: 'no'
-          PKG_CONFIG_SYSROOT_DIR: '@BUILD_ROOT@/system-root'
-          PKG_CONFIG_LIBDIR: '@BUILD_ROOT@/system-root/usr/lib/pkgconfig:@BUILD_ROOT@/system-root/usr/share/pkgconfig'
+        - '--libdir=lib'
+        - '--buildtype=debugoptimized'
+        - '-Dxlib-xcb=enabled'
+        - '-Dzlib=enabled'
+        - '-Dglib=enabled'
+        - '-Dtests=disabled'
```
[the remark this answers is not in the document]
Comment (Dennisbonke) [reply bootstrap.d/x11-libs.yml:83]: `- '-Dipc_rmid_deferred_release=false'` seems to be needed, I didn't check which value is correct. However, I can't get meson to take the option. I propose holding cairo, which is in line with BLFS.

**Review by Dennisbonke — CHANGES_REQUESTED** [review]
A bunch of things don't quite work for me, nor your testcase with `xclock`. While I'll admit that `xclock` was broken locally before, it still doesn't work for me, which is something to investigate. For now, here are a list of remarks, we can and should discuss them over the next few days if you have any questions.

**Comment by ElectrodeYT** [comment]
> A bunch of things don't quite work for me, nor your testcase with `xclock`. While I'll admit that `xclock` was broken locally before, it still doesn't work for me, which is something to investigate. For now, here are a list of remarks, we can and should discuss them over the next few days if you have any questions.

Extremely interesting, I have some other local changes which I will investigate to see if they are the cause of making xwayland apps work.
The fact that so many ports didn't work is curious, as they all did locally: it's possible i just forgot to nuke their source directories fully, but that seems unlikely. I'll check later when I'm back home

File: bootstrap.d/x11-libs.yml [hunk 5]
```diff
@@ -1580,6 +1552,7 @@ packages:
         environ:
           DESTDIR: '@THIS_COLLECT_DIR@'
 
+          # TODO: newer autoconf
```
[the remark this answers is not in the document]
Comment (ElectrodeYT) [reply bootstrap.d/x11-libs.yml:1555]: libxt did not compile, it wanted a newer autoconf. See above comment on updating autoreconf

File: bootstrap.d/x11-libs.yml [hunk 6]
```diff
@@ -1642,7 +1615,6 @@ packages:
       - xorg-util-macros
       - xorg-proto
       - libxcb
-    revision: 5
```
[the remark this answers is not in the document]
Comment (ElectrodeYT) [reply bootstrap.d/x11-libs.yml:1645]: See above; will fix this later.

File: bootstrap.d/x11-libs.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (ElectrodeYT) [reply bootstrap.d/x11-libs.yml:83]: > `- '-Dipc_rmid_deferred_release=false'` seems to be needed, I didn't check which value is correct. However, I can't get meson to take the option. I propose holding cairo, which is in line with BLFS.

I forgot the patch file for this: this is supposed to go into the meson native-file, but it did not work either, so I patched cairo to "fix" this. This addition to cairo is quite new, and it looks like it doesn't work for everyone either, so I'm pretty sure its going to be fixed upstream at some point soon anyway.

File: bootstrap.d/x11-libs.yml [hunk 7]
```diff
@@ -67,37 +66,35 @@ packages:
...
-        quiet: true
 
   - name: gdk-pixbuf
     architecture: '@OPTION:arch@'
     source:
       subdir: ports
       git: 'https://gitlab.gnome.org/GNOME/gdk-pixbuf.git'
-      tag: '2.40.0'
-      version: '2.40.0'
+      tag: '2.42.10'
```
[the remark this answers is not in the document]
Comment (ElectrodeYT) [reply bootstrap.d/x11-libs.yml:96]: Will do

File: bootstrap.d/x11-libs.yml (hunk 1 above)
Comment (ElectrodeYT) [reply bootstrap.d/x11-libs.yml:500]: I propose updating autoconf instead

File: bootstrap.d/x11-misc.yml [hunk 8]
```diff
@@ -116,8 +117,8 @@ packages:
     source:
       subdir: 'ports'
       git: 'https://gitlab.freedesktop.org/xkeyboard-config/xkeyboard-config.git'
-      tag: 'xkeyboard-config-2.37'
-      version: '2.37'
+      tag: 'xkeyboard-config-2.39'
```
[the remark this answers is not in the document]
Comment (ElectrodeYT) [reply bootstrap.d/x11-misc.yml:118]: Done

File: bootstrap.d/x11-misc.yml [hunk 9]
```diff
@@ -79,8 +80,8 @@ packages:
     source:
       subdir: ports
       git: 'https://gitlab.freedesktop.org/xorg/data/bitmaps.git'
-      tag: 'xbitmaps-1.1.2'
-      version: '1.1.2'
+      tag: 'xbitmaps-1.1.3'
```
[the remark this answers is not in the document]
Comment (ElectrodeYT) [reply bootstrap.d/x11-misc.yml:82]: Done

**Review by Dennisbonke — APPROVED** [review]
LGTM now

**Pull request merged**